### ISSUE TITLE
[2.5.3] [ubuntu 20.04] [bug] running 'task overdue' seems to have a memory leak and locks up cpu

### ISSUE THREAD

**Issue by arooni** [GothenburgBitFactory/taskwarrior#2389]
#### To report a bug...

* What command(s) did you run?
task overdue
also occurs with " task rc.context:default overdue"

* What did you expect to happen?
the command to run as normal
* What actually happened?

computer locks up and i have to kill the task

![image](https://user-images.githubusercontent.com/556667/104820630-d1036300-57fb-11eb-829c-75e2f63da4b7.png)

**Comment by arooni** [comment]
gets to 100% cpu as well as doesnt seem to finish.  do i really have *that many* overdue tasks? :upside_down_face:

**Comment by tbabej** [comment]
@arooni We'll need to see your tasks in order to get to root of this :) ideally you could narrow down your `pending.data` and/or `completed.data` to the subset of tasks that are causing this.

Does this only happen with `task overdue` or also other reports (i.e. `task all`)?

**Comment by arooni** [comment]
Yes it happens with task all.  Might I have a terrible chunky task that's causing all of this?

**Comment by arooni** [comment]
happens with 

task overdue
task completed

**Comment by arooni** [comment]
pretty sure it was the # of dependencies on this task

```
[depends:"bc6612d3-8b28-4a5f-ab5f-9cb0bc5eeffb,cddeed62-df01-4fee-9107-136de95d425d,463e3676-4aa4-486e-81f4-b78a60ac4cc6,709b299d-2951-41d7-b2c3-293ad13316fb,87aaf099-8915-4c22-bfb0-7067d6cc7c03,82334998-eff5-42e8-8ad6-fdbaeffa65db,d8a3f704-624a-4001-a3a3-2841c62cb52c,54351e9d-c097-445e-93e9-4e42e0c11f49,0951a914-db7d-447b-9f33-886587379ecb,5db8f66d-888b-47b9-b506-85a991010c8d,943b67ca-d397-4996-9311-108ea6d997a0,dc8882d0-6dc9-44a5-9901-3c4c4a6f71cd,abe37633-2a75-44d6-813c-c2b5fb8a54fe" description:"ensure that flow on projectx still works" due:"1584248400" entry:"1571364059" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99ebfc8157879d9faa438" intheamtrellolastupdated:"2020-06-17T22:11:06.613Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"https:\/\/trello.com\/c\/gJubF6LG\/431-ensure-that-flow-project-x-still-works" modified:"1592431882" project:"alabs.projectx" scheduled:"1571806800" status:"pending" totalactivetime:"4033seconds" uuid:"98be4ce5-95b8-4c46-a5cf-adf03abb6092"]


```

**Comment by tbabej** [comment]
@arooni Can you then export these including the dependencies' UUIDs? Also possible to share privately, if needed.

**Comment by arooni** [comment]
i removed all the tasks from pending.data that had a bunch of dependencies; seemed to clear it up

**Comment by arooni** [comment]
i could share them privately ; how would you like me to send them?

**Comment by arooni** [comment]
i grabbed all the dependencies and sanitized the output for public consumption

```
[depends:"bc6612d3-8b28-4a5f-ab5f-9cb0bc5eeffb,cddeed62-df01-4fee-9107-136de95d425d,463e3676-4aa4-486e-81f4-b78a60ac4cc6,709b299d-2951-41d7-b2c3-293ad13316fb,87aaf099-8915-4c22-bfb0-7067d6cc7c03,82334998-eff5-42e8-8ad6-fdbaeffa65db,d8a3f704-624a-4001-a3a3-2841c62cb52c,54351e9d-c097-445e-93e9-4e42e0c11f49,0951a914-db7d-447b-9f33-886587379ecb,5db8f66d-888b-47b9-b506-85a991010c8d,943b67ca-d397-4996-9311-108ea6d997a0,dc8882d0-6dc9-44a5-9901-3c4c4a6f71cd,abe37633-2a75-44d6-813c-c2b5fb8a54fe" description:"ahha" due:"1584248400" entry:"1571364059" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99ebfc8157879d9faa438" intheamtrellolastupdated:"2020-06-17T22:11:06.613Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:";;" modified:"1592431882" project:"" scheduled:"1571806800" status:"pending" totalactivetime:"4033seconds" uuid:"98be4ce5-95b8-4c46-a5cf-adf03abb6092"]
[depends:"6c877233-fb82-485d-9ef9-d79c468eaf6a,d5fe546d-df40-4e04-962c-a581a4d7cbc5,4a15fb93-c175-41a6-84c6-c39ed5f0e805,644344b7-f269-434b-9293-68eb810ddb2f" description:"blah" due:"1584248400" entry:"1576280527" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f009fe4ea258e772b6f" intheamtrellolastupdated:"2020-06-05T02:19:07.441Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323558" project:"" status:"pending" uuid:"4b273543-dbbe-42d8-b77a-6fe363e76512"]
[depends:"db34ed08-1dc1-4a38-a626-63ad07146ed1" description:"blahblah" due:"1584248400" entry:"1576281803" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f0a520b9f571ded2e65" intheamtrellolastupdated:"2020-06-05T02:19:02.505Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323614" project:"" scheduled:"1579500000" status:"pending" uuid:"d649e19c-ac25-4859-a955-480bd591eb17"]
[depends:"fed606fa-e2af-45b9-bb53-aa72b58a6ee0" description:"blahbalbahl" entry:"1579654050" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f5012046b86d33e9b94" intheamtrellolastupdated:"2020-06-05T02:18:24.798Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323509" status:"pending" uuid:"d86904ce-686f-432e-8c77-076a7fe03236"]
[annotation_1582071321:"test" depends:"abc1f29b-f1c0-4b9e-a4f1-0acd6af40688,94d9354d-2c52-498d-ba34-20d24893db84,1182d6ec-1b75-4390-80d9-092880a88632,d86904ce-686f-432e-8c77-076a7fe03236,3ea01a5e-1813-484e-9169-9aa0237b924a,279dfc63-5125-4526-a933-984f6171f71b,ceb8c62c-ca34-4d10-94ef-ba4e3edce0a4,be424059-d704-40d2-a0bf-6a74ecf2a9a6,f30a7b72-f92b-4859-bd7f-1c8bffacca2f" description:"blablhablh" due:"1584248400" entry:"1579655015" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f5951c66480080cfb18" intheamtrellolastupdated:"2020-06-05T02:18:21.667Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323562" project:"" status:"pending" uuid:"e236c4df-33b4-43e0-ac37-c991f1a4f6ce"]
[depends:"bc6612d3-8b28-4a5f-ab5f-9cb0bc5eeffb,cddeed62-df01-4fee-9107-136de95d425d,463e3676-4aa4-486e-81f4-b78a60ac4cc6,709b299d-2951-41d7-b2c3-293ad13316fb,87aaf099-8915-4c22-bfb0-7067d6cc7c03,82334998-eff5-42e8-8ad6-fdbaeffa65db,d8a3f704-624a-4001-a3a3-2841c62cb52c,54351e9d-c097-445e-93e9-4e42e0c11f49,0951a914-db7d-447b-9f33-886587379ecb,5db8f66d-888b-47b9-b506-85a991010c8d,943b67ca-d397-4996-9311-108ea6d997a0,dc8882d0-6dc9-44a5-9901-3c4c4a6f71cd,abe37633-2a75-44d6-813c-c2b5fb8a54fe" description:"asldkjfklasdjafsjdkl" due:"1584248400" entry:"1571364059" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99ebfc8157879d9faa438" intheamtrellolastupdated:"2020-06-17T22:11:06.613Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1592431882" project:"" scheduled:"1571806800" status:"pending" totalactivetime:"4033seconds" uuid:"98be4ce5-95b8-4c46-a5cf-adf03abb6092"]
[depends:"6c877233-fb82-485d-9ef9-d79c468eaf6a,d5fe546d-df40-4e04-962c-a581a4d7cbc5,4a15fb93-c175-41a6-84c6-c39ed5f0e805,644344b7-f269-434b-9293-68eb810ddb2f" description:"aklfsdfjlkdsa" due:"1584248400" entry:"1576280527" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f009fe4ea258e772b6f" intheamtrellolastupdated:"2020-06-05T02:19:07.441Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323558" project:"" status:"pending" uuid:"4b273543-dbbe-42d8-b77a-6fe363e76512"]
[depends:"db34ed08-1dc1-4a38-a626-63ad07146ed1" description:"aklsdfklasdjlasd" due:"1584248400" entry:"1576281803" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f0a520b9f571ded2e65" intheamtrellolastupdated:"2020-06-05T02:19:02.505Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323614" project:"" scheduled:"1579500000" status:"pending" uuid:"d649e19c-ac25-4859-a955-480bd591eb17"]
[depends:"fed606fa-e2af-45b9-bb53-aa72b58a6ee0" description:"laksjfklasdjfklasdj" entry:"1579654050" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f5012046b86d33e9b94" intheamtrellolastupdated:"2020-06-05T02:18:24.798Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323509" status:"pending" uuid:"d86904ce-686f-432e-8c77-076a7fe03236"]
[annotation_1582071321:"test" depends:"abc1f29b-f1c0-4b9e-a4f1-0acd6af40688,94d9354d-2c52-498d-ba34-20d24893db84,1182d6ec-1b75-4390-80d9-092880a88632,d86904ce-686f-432e-8c77-076a7fe03236,3ea01a5e-1813-484e-9169-9aa0237b924a,279dfc63-5125-4526-a933-984f6171f71b,ceb8c62c-ca34-4d10-94ef-ba4e3edce0a4,be424059-d704-40d2-a0bf-6a74ecf2a9a6,f30a7b72-f92b-4859-bd7f-1c8bffacca2f" description:"" due:"1584248400" entry:"1579655015" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f5951c66480080cfb18" intheamtrellolastupdated:"2020-06-05T02:18:21.667Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323562" project:"" status:"pending" uuid:"e236c4df-33b4-43e0-ac37-c991f1a4f6ce"]


```

**Comment by arooni** [comment]
hmmm removed all these guys but still running into the problem on task +overdue; will keep lookin

**Comment by arooni** [comment]
```

[annotation_1589419350:"" description:"" due:"1593579600" entry:"1586568813" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99f97f11c2d57c44c7c5a" intheamtrellolastupdated:"2020-06-05T02:17:45.649Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"" modified:"1591323471" project:"self.improvement" status:"pending" tags:"focus,monthlygoal" uuid:"91ead579-d99f-4edb-8e48-a578ffb79de9"]


```

when i removed this item plus the depends i think it fixed it

**Comment by arooni** [comment]
i think the issue was *realllly* long annotations ; or perhaps characters like * the bullet point in unicode.  this fixes the task overdue but not the task all; there are other annotations that are causing this

here's an example; 

```
[annotation_1523918233:"•    Do morning success ritual as written; Do evening success ritual as written; 3 tasks per day on my task list * Use countdown timers to ensure that my meal times fit within margins.. don't count UP.  count down" annotation_1526941323:"•    Try long walk meditations where instead of consuming content I think about outlining articles or business ideas concepts.  A.la deep work Try biking + listening to interesting content; same for a long walk." annotation_1528241513:"try working on ONLY what is assigned until its complete" annotation_1528757245:"this weekend: pick a fun side project and do something with it." annotation_1528757764:"stop working each night @ 7PM and actually stop" annotation_1531870039:"complete focus each week" annotation_1532453571:"•Trying blahblablhablh." annotation_1535241971:"try breaking up my day to play basketball after 3-4 hours of productivity" annotation_1536016960:"Max 3 tasks for each work day and complete each of them before adding more" annotation_1543799537:"deep meaningful breathing throughout the day (box breathing 4\/7\/8 etc)" description:"A\/B Test: Work Techniques" due:"1584248400" entry:"1523918222" estimate:"PT10M" intheamtrelloboardid:"5ed999522e23473defafe355" intheamtrelloid:"5ed99b6b2537837e3ea76f55" intheamtrellolastupdated:"2020-06-17T22:10:45.372Z" intheamtrellolistid:"5ed999522e23473defafe356" intheamtrellolistname:"To Do" intheamtrellourl:"https:\/\/trello.com\/c\/9OUCWD9X\/150-a-b-test-work-techniques" modified:"1592431910" project:"self.improvement" status:"pending" tags:"abtest,shallow" uuid:"52b33e3e-3983-4008-9f6c-464a68d15128"]

```

**Comment by arooni** [comment]
update: removed all the items with the * unicode and removed really long annoations from completed and now it seems to work ok;

**Comment by tbabej** [comment]
I am trying to reproduce but was unfortunately not able to reproduce this on 2.5.3, using the task you pasted above.

**Comment by arooni** [comment]
i can share you the one annotation line in completed.data that causes this to occur for me but would rather do so privately

**Comment by tbabej** [comment]
It seems I was successful with reproducing, it seems the reproducibility depends on the size of the terminal window (I cannot reproduce with full-screen terminal window (140 columns), but with just 70 columns I can indeed see `task` hang.

**Comment by arooni** [comment]
i can reproduce with the right annotation in completed.txt; its quite long though; not sure how best to share it with you

**Comment by tbabej** [comment]
No need to share, as I already have a reproducer. This is likely related to some text wrapping problem - the size of the terminal under which the issue will manifest will depend on the length of the annotation.

**Comment by arooni** [comment]
awesome!  let me know if you need more help!

**Comment by tbabej** [comment]
Fixed in 2.6.0.
